lib/SILOptimizer/Transforms/SILCodeMotion.cpp

```diff
@@ -1759,6 +1787,36 @@ static bool processFunction(SILFunction *F, AliasAnalysis *AA,
     LLVM_DEBUG(llvm::dbgs() << "    Performing the dataflow!\n");
     Changed |= State.process();
   }
+  
+  if (strCmpInsts.size() > 3 /* some threshold */) {
+    std::vector<SILValue> findStrCaseVals;
+    for (auto& inst : strCmpInsts) {
+      // The first argument is the case value
+      SILFunction *fn = inst->getInitiallyReferencedFunction();
+      if (fn && !fn->empty())
+    		findStrCaseVals.push_back(fn->getArguments().front());
+    }
+    
+    // TODO: turn the findStrCaseVals into an array
+    
+    // Find the function decl
+    SmallVector<ValueDecl *, 1> results;
+    Builder.getASTContext().lookupInSwiftModule("_findStringSwitchCase", results);
```

Comment:
Like I said earlier, you can put this function in KnownDecls.def that way it can be cached.

Reply:
Will do. Will that fix the `lookUpFunction ` issue?

Reply:
What you probably want to do is call something like `emitApplyOfLibraryIntrinsic` once you get the func decl from the astcontext.

Reply:
Thanks, @Azoy, that is super helpful. Looks like `emitApplyOfLibraryIntrinsic` is a member of `SILGenFunction`. How can I get access to that in the SILOptimizer?

Reply:
Oh my apologies. You’re correct that this is only available in SILGen (whoops). What you can do however is something like the following:
```cpp
auto mangled = SILDeclRef(findFn, SILDeclRef::Kind::Func).mangle();
auto *silFn = F->getModule().findFunction(mangled, SILLinkage::PublicExternal);
```

Reply:
Thanks, I really appreciate your help. Unfortunately, that still gave me a `nullptr`. Any other ideas? I've looked through some other places `findFunction` is used in the SILOptimizer and it seems like it _should_ work but, it doesn't for some reason. Maybe we can hardcode the mangled function (like what's done for `_swift_isClassOrObjCExistentialType`)?